libglusterfs/src/glusterfs/mem-pool.h

```diff
@@ -158,6 +158,18 @@ __gf_default_realloc(void *oldptr, size_t size)
 
 #define GF_FREE(free_ptr) __gf_free(free_ptr)
 
+#if defined(GF_DISABLE_MEMPOOL)
+#define mem_get(mem_pool)                                                      \
+    GF_MALLOC((unsigned long)mem_pool, gf_common_mt_mem_pool);
```

Comment:
I would add a comment to explain how the memory pool object pointer is reused to track the object size. This is a very useful and cheap method, but it's a bit hacky, so it deserves a comment.

Reply:
done